### PR TITLE
include line:column in error messages

### PR DESCRIPTION
**Pull request by martine**
Track the current ts.Node while traversing, and include the
offset of that node when failing an assertion.  This isn't
great at finding the correct point but it is at least close.

### CODE REVIEW

File: src/sickle.ts [hunk 1]
```diff
@@ -259,17 +266,20 @@ class Annotator {
...
-    return this.writeRange(from, to);
+    return this.writeRange(from, node.getFullStart());
   }
 
-  private fail(msg: string) { throw new Error(msg); }
-}
-
-function assert(condition: boolean, msg: string) {
-  if (!condition) throw new Error(msg);
+  private fail(msg: string) {
```
Comment (mprobst) [inline src/sickle.ts]: Random drive by review comment: maybe compare with this code here, where we solved the problem previously?

https://github.com/angular/ts2dart/blob/master/lib/main.ts#L241

Comment (martine) [inline src/sickle.ts]: That is way better.  I will send an update when this is ready for re-review.

**Comment by martine** [comment]
PTAL

File: src/sickle.ts [hunk 2]
```diff
@@ -261,15 +265,19 @@ class Annotator {
...
+    this.assert(to > from, `Offset must not be smaller; ${to} vs ${from}`);
     return this.writeRange(from, to);
   }
 
-  private fail(msg: string) { throw new Error(msg); }
-}
+  private fail(msg: string) {
+    let offset = this.currentNode.getFullStart();
+    let {line, character} = this.file.getLineAndCharacterOfPosition(offset);
+    throw new Error(`near node starting at ${line+1}:${character+1}: ${msg}`);
```
Comment (rkirov) [inline src/sickle.ts]: I was thinking whether we should add a test for this, but I don't see yet a failure / assertion that would be useful to the external user of the tool. All the errors we have are inconsistencies in our internal state, as which point you can argue it is not worth the overhead of testing that. What do you think?

**Comment by rkirov** [comment]
LGTM

**Comment by mprobst** [comment]
LGTM.

**Pull request merged**